lib/srv/desktop/tdp/proto.go

```diff
@@ -241,22 +333,14 @@ type MouseMove struct {
 func (m MouseMove) Encode() ([]byte, error) {
 	buf := new(bytes.Buffer)
 	buf.WriteByte(byte(TypeMouseMove))
-	if err := binary.Write(buf, binary.BigEndian, m); err != nil {
-		return nil, trace.Wrap(err)
-	}
+	writeUint32(buf, m.X)
+	writeUint32(buf, m.Y)
 	return buf.Bytes(), nil
 }
 
 func decodeMouseMove(in peekReader) (MouseMove, error) {
```

Comment:
most of these functions no longer require a `peekReader`, just an `io.Reader` would be fine, right? Not only would it be more correct to update these to just `io.Reader`, I think this would be much easier to reason about and review this change if `peekReader` was removed or renamed, so that the compiler would enforce that all current references to `peekReader` are updated to avoid mistakenly calling `ReadByte` (now that we no longer call `UnreadByte` above)

Reply:
Changed most of them to `io.Reader`.

We could get rid of `peekReader` entirely, though it would be a little bit annoying to read a single byte via `io.Reader.Read` anywhere we need a byte at a time.

Reply:
yeah it's alright, I suggested it mostly to guarantee we didn't forget to update any `decode` functions because I'm not extremely familiar with this part of the code, but I'm happy to trust you

I think `peek` often implies you can do something like `Peek()` or `UnreadByte` where you can "peek" at something without affecting further reads, maybe changing the name to `byteReader` would kind of avoid using that and also guarantee we don't miss any `peekReader` users here? Feel free to ignore if you disagree, I will approve once the lint is fixed

Reply:
I agree with changing the name. Will do. We used to actually read and unread a byte and I removed that because it felt silly to validate that same byte and the top level and in each individual decode func.